packages/components/src/systems/SignatureAttestation/SignatureAttestationModal.tsx

```diff
@@ -0,0 +1,252 @@
+  query GetCurrentUserSignatureAttestationStatus {
+    me {
+      signatureAttestationStatus {
+        ... on NeedsSignatureAttestation {
+          version
+          content
+        }
+        ... on CompletedSignatureAttestation {
+          agreedAt
+          version
+        }
+        ... on NotApplicableSignatureAttestation {
+          reason
+        }
+      }
+    }
+  }
+`;
+
+export type GetCurrentUserSignatureAttestationStatusType = {
+  me?: {
+    signatureAttestationStatus?:
+      | {
+          __typename: 'NeedsSignatureAttestation';
+          version: string;
+          content: string;
+        }
+      | {
+          __typename: 'CompletedSignatureAttestation';
+          agreedAt: Date;
+          version: string;
+        }
+      | {
+          __typename: 'NotApplicableSignatureAttestation';
+          reason: string | null | undefined;
+        };
+  };
+};
+
+const AgreeToSignatureAttestation = gql`
+  mutation AgreeToSignatureAttestation($version: String!) {
+    agreeToSignatureAttestation(version: $version)
+  }
+`;
+
+export type AgreeToSignatureAttestationType = {
+  agreeToSignatureAttestation?: boolean;
+};
+
+type DataReturn<Type> = {
+  data?: Type;
+  loading?: boolean;
+  error?: any;
+  errors?: readonly any[];
+};
+
+const getCurrentUserSignatureAttestationStatus = async (
+  client: ApolloClient<any>
+): Promise<DataReturn<GetCurrentUserSignatureAttestationStatusType>> => {
+  return await client.query<GetCurrentUserSignatureAttestationStatusType>({
+    query: GetCurrentUserSignatureAttestationStatus,
+    fetchPolicy: 'cache-first'
+  });
+};
+
+const agreeToSignatureAttestation =
+  (client: ApolloClient<any>) =>
+  async (version: string): Promise<DataReturn<Maybe<AgreeToSignatureAttestationType>>> => {
+    return await client.mutate<AgreeToSignatureAttestationType, { version: string }>({
+      mutation: AgreeToSignatureAttestation,
+      variables: { version },
+      refetchQueries: ['GetCurrentUserSignatureAttestationStatus'],
+      awaitRefetchQueries: true,
+      update: (cache, { data }) => {
+        if (data?.agreeToSignatureAttestation) {
+          cache.writeQuery({
+            query: GetCurrentUserSignatureAttestationStatus,
+            data: {
+              me: {
+                signatureAttestationStatus: {
+                  __typename: 'CompletedSignatureAttestation',
+                  agreedAt: new Date(),
+                  version
+                }
+              }
+            }
+          });
+        }
+      }
+    });
+  };
+
+export interface SignatureAttestationModalProps {
+  client: ApolloClient<any>;
+  children: JSXElement;
+  onAgreeComplete?: () => void;
+}
+
+type Status =
+  | {
+      status: 'LOADING';
+    }
+  | { status: 'NEEDS ATTESTATION'; version: string }
+  | { status: 'COMPLETE' }
+  | { status: 'ERROR'; errors: any[] };
+
+export const SignatureAttestationModal = (props: SignatureAttestationModalProps) => {
```

Comment:
nit on name, call it `SignatureAttestationCard`

Reply:
Its a modal (takes over the whole area) not just a card